src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -189,7 +185,18 @@ namespace ts.codefix {
             }
         });
 
-        return getSynthesizedDeepClone(nodeToRename, /*includeTrivia*/ true, synthNamesMap, lastDotThenMap, checker, originalType);
+        function isCallbackOnTypePromise(child: Node, setOfAllCallbacksToReturn: Map<true>): boolean {
+            let node = child.parent;
+            if (isCallExpression(node) || isIdentifier(node) && !setOfAllCallbacksToReturn.get(getNodeId(node).toString())) {

```

Comment:
I can't tell whether you intended the `setOfAllCallbacksToReturn` check to apply in the call expression case.